Database update error after updating to lnd v0.5-beta-rc2

This is the error I get during the start process:

2018-09-10 16:32:14.944 [INF] LTND: Version 0.5.0-beta commit=1941353fb28755a170793e43595601d75c8f3dda
2018-09-10 16:32:14.944 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2018-09-10 16:32:14.944 [INF] CHDB: Checking for schema update: latest_version=6, db_version=5
2018-09-10 16:32:14.944 [INF] CHDB: Performing database schema migration
2018-09-10 16:32:14.944 [INF] CHDB: Applying migration #6
2018-09-10 16:32:14.945 [INF] CHDB: Migrating database to properly prune edge update index
2018-09-10 16:32:14.964 [INF] LTND: Shutdown complete
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*leafPageElement).value(...)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/page.go:126
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).Stats.func1(0x7f344127a000, 0x3)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:440 +0x34a
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc420590000, 0x527a, 0x3, 0xc4206d3660)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:607 +0x81
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc420590000, 0x4d23, 0x2, 0xc4206d3660)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc420590000, 0x1942e, 0x1, 0xc4206d3660)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc420590000, 0x11c43, 0x0, 0xc4206d3660)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).forEachPage(0xc4200bb700, 0xc4206d3660)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:486 +0x77
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).Stats(0xc4200bb700, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:403 +0x131
github.com/lightningnetwork/lnd/channeldb.migratePruneEdgeUpdateIndex(0xc420590000, 0xfad16b, 0x16)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/channeldb/migrations.go:518 +0x2bc
github.com/lightningnetwork/lnd/channeldb.(*DB).syncVersions.func1(0xc420590000, 0x110df48, 0xc420590000)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/channeldb/db.go:724 +0x14f
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*DB).Update(0xc420588000, 0xc42045fa18, 0x0, 0x0)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/db.go:668 +0x90
github.com/lightningnetwork/lnd/channeldb.(*DB).syncVersions(0xc4201c6060, 0x1774d60, 0x7, 0x7, 0xc420588000, 0x0)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/channeldb/db.go:716 +0x34f
github.com/lightningnetwork/lnd/channeldb.Open(0xc4200381b0, 0x26, 0x3, 0xc4200381b0, 0x26)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/channeldb/db.go:124 +0x184
main.lndMain(0x0, 0x0)
	/home/peernode/go/src/github.com/lightningnetwork/lnd/lnd.go:168 +0x3fc
main.main()
	/home/peernode/go/src/github.com/lightningnetwork/lnd/lnd.go:455 +0x26

Please help. I tried with the official precompilled release. The result is the same.

This is my attempt to run the compiled version:

v./lnd
2018-09-10 16:34:49.878 [INF] LTND: Version 0.5.0-beta commit=fb95858afce6d7129758e7677bcb1552c5be2c51
2018-09-10 16:34:49.878 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2018-09-10 16:34:49.879 [INF] CHDB: Checking for schema update: latest_version=6, db_version=5
2018-09-10 16:34:49.879 [INF] CHDB: Performing database schema migration
2018-09-10 16:34:49.879 [INF] CHDB: Applying migration #6
2018-09-10 16:34:49.879 [INF] CHDB: Migrating database to properly prune edge update index
2018-09-10 16:34:49.900 [INF] LTND: Shutdown complete
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*leafPageElement).value(...)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/page.go:126
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).Stats.func1(0x7f17b4790000, 0x3)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:440 +0x342
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc0004680e0, 0x527a, 0x3, 0xc000065690)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:607 +0x81
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc0004680e0, 0x4d23, 0x2, 0xc000065690)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc0004680e0, 0x1942e, 0x1, 0xc000065690)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Tx).forEachPage(0xc0004680e0, 0x11c43, 0x0, 0xc000065690)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/tx.go:613 +0xd7
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).forEachPage(0xc0000a6e00, 0xc000451690)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:486 +0x77
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*Bucket).Stats(0xc0000a6e00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/bucket.go:403 +0x13f
github.com/lightningnetwork/lnd/channeldb.migratePruneEdgeUpdateIndex(0xc0004680e0, 0xfa9aee, 0x16)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/channeldb/migrations.go:518 +0x2bf
github.com/lightningnetwork/lnd/channeldb.(*DB).syncVersions.func1(0xc0004680e0, 0x110b4a0, 0xc0004680e0)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/channeldb/db.go:724 +0x148
github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*DB).Update(0xc0004c0000, 0xc000451a30, 0x0, 0x0)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/db.go:668 +0x90
github.com/lightningnetwork/lnd/channeldb.(*DB).syncVersions(0xc000094260, 0x19d87e0, 0x7, 0x7, 0xc0004c0000, 0x0)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/channeldb/db.go:716 +0x335
github.com/lightningnetwork/lnd/channeldb.Open(0xc0000363f0, 0x26, 0x3, 0xc0000363f0, 0x26)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/channeldb/db.go:124 +0x178
main.lndMain(0x0, 0x0)
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/lnd.go:168 +0x3f8
main.main()
	/Users/roasbeef/gocode/src/github.com/lightningnetwork/lnd/lnd.go:455 +0x26

--- a/channeldb/migrations.go
+++ b/channeldb/migrations.go
@@ -473,12 +473,9 @@ func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
 	if edges == nil {
 		return nil
 	}
-	edgeUpdateIndex, err := edges.CreateBucketIfNotExists(
-		edgeUpdateIndexBucket,
-	)
-	if err != nil {
-		return fmt.Errorf("unable to create/fetch edge update " +
-			"index bucket")
+	edgeUpdateIndex := edges.Bucket(edgeUpdateIndexBucket)
+	if edgeUpdateIndex == nil {
+		return nil
 	}
 
 	// Retrieve some buckets that will be needed later on. These should
@@ -518,7 +515,6 @@ func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
 
 	// With the existing edge policies gathered, we'll recreate the index
 	// and populate it with the correct entries.
-	oldNumEntries := edgeUpdateIndex.Stats().KeyN
 	if err := edges.DeleteBucket(edgeUpdateIndexBucket); err != nil {
 		return fmt.Errorf("unable to remove existing edge update "+
 			"index: %v", err)
@@ -557,10 +553,6 @@ func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
 			return err
 		}
 	}
-
-	newNumEntries := edgeUpdateIndex.Stats().KeyN
-	log.Infof("Pruned %d stale entries from the edge update index",
-		oldNumEntries-newNumEntries)
 
 	log.Info("Migration to properly prune edge update index complete!")
 

